protocols/mdns/src/query.rs

```diff
@@ -303,3 +290,50 @@ impl fmt::Debug for MdnsPeer {
             .finish()
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::dns::build_query_response;
+
+    #[test]
+    fn test_create_mdns_peer() {
+        let ttl = 300;
+        let peer_id = PeerId::random();
+
+        let mut addr1: Multiaddr = "/ip4/1.2.3.4/tcp/5000".parse().unwrap();
+        let mut addr2: Multiaddr = "/ip6/::1/udp/10000".parse().unwrap();
+        addr1.push(Protocol::P2p(peer_id.clone().into()));
+        addr2.push(Protocol::P2p(peer_id.clone().into()));
+
+        let packets = build_query_response(
+            0xf8f8,
+            peer_id,
+            vec![addr1, addr2].into_iter(),
+            Duration::from_secs(60),
+        );
+
+        for bytes in packets {
+            let packet = Packet::parse(&bytes).unwrap();
+            let record_value = packet
+                .answers
+                .iter()
+                .filter_map(|record| {
+                    if record.name.to_string().as_bytes() != SERVICE_NAME {
+                        return None;
+                    }
+                    let record_value = match record.data {
+                        RData::PTR(record) => record.0.to_string(),
+                        _ => return None,
+                    };
+                    return Some(record_value);
+                })
+                .next()
+                .unwrap();
+
+            let peer = MdnsPeer::new(&packet, record_value, ttl).unwrap();
```

Comment:
nit, I would make this an `expect` so that if this fails it's a useful message to the test runner.